pkgdb/src/buildenv/realise.cc

```diff
@@ -605,6 +609,61 @@ getRealisedPackages( nix::ref<nix::EvalState> &         state,
 }
 
 
+/* -------------------------------------------------------------------------- */
+
+void
+addScriptToScriptsDir( const std::string &           scriptContents,
+                       const std::filesystem::path & scriptsDir,
+                       const std::string &           scriptName,
+                       std::stringstream & mainActivationScriptContents,
+                       const bool          shouldSource )
+{
+  /* Ensure that the "activate" subdirectory exists. */
+  std::filesystem::create_directories( scriptsDir / ACTIVATION_SUBDIR_NAME );
+
+  /* Write the script to a temporary file. */
+  std::filesystem::path scriptTempPath( nix::createTempFile().second );
+  debugLog(
+    nix::fmt( "created tempfile for activation script: script=%s, path=%s",
+              scriptName,
+              scriptTempPath ) );
+  std::ofstream scriptTmpFile( scriptTempPath );
+  if ( ! scriptTmpFile.is_open() )
+    {
+      throw ActivationScriptBuildFailure( std::string( strerror( errno ) ) );
+    }
+  scriptTmpFile << scriptContents << std::endl;
+  if ( scriptTmpFile.fail() )
+    {
+      throw ActivationScriptBuildFailure( std::string( strerror( errno ) ) );
+    }
+  scriptTmpFile.close();
+
+  /* Copy the script to the scripts directory. */
+  auto scriptPath = scriptsDir / ACTIVATION_SUBDIR_NAME / scriptName;
+  debugLog( nix::fmt( "copying script to scripts dir: src=%s, dest=%s",
+                      scriptTempPath,
+                      scriptPath ) );
+  std::filesystem::copy_file( scriptTempPath, scriptPath );
+  std::filesystem::permissions( scriptPath,
+                                std::filesystem::perms::owner_exec,
+                                std::filesystem::perm_options::add );
+
+  /* Reference the script from the main activation script. */
+  auto scriptRelativePath
+    = nix::fmt( "\"$FLOX_ENV/%s/%s\"", ACTIVATION_SUBDIR_NAME, scriptName );
```

Comment:
nit/question nonblocking: why are you using the env var rather than an absolute path in the /nix/store?